DirectoryCache: use custom comparator.

hanetzer opened the pull request:
I think this fixes #6783 , user pupnewfster on discord showed only the
line-endings changing when they ran it.

marchermans commented on src/test/java/net/minecraftforge/debug/world/MarkDimensionForDeletionTest.java:
```diff
@@ -61,7 +61,7 @@
             if (!worldIn.isRemote)
             {
-                if (playerIn.isShiftKeyDown())
+                if (playerIn.isSneaking())
```
Why was this additional change made in this PR?

hanetzer replied:
because mapping changes made the test not run.

LexManos commented:
Your test means nothing. It wouldnt ever expose anything while run on a the same os over and over.
As for your solution, as I said in the linked issue, sorting the stream while writing is the best way to keep it localilized patch wise. It'd even let us delete our TreeMap change.
It just needs `.sorted((a,b) -> a.split(" ")[1].compareTo(b.split(" ")[1])`

hanetzer commented:
> Your test means nothing. It wouldnt ever expose anything while run on a the same os over and over.

It seemed like this would be the same kind of sorting issue as mentioned in the original issue,
a top-level tag foo**s**.json and nested tags under foo/stuff.json, but I guess I was
mistaken. I did have pup try it on their windows machine. Could have someone test it without
the comparator constructor on windows as well.

> As for your solution, as I said in the linked issue, sorting the stream while writing is the best way to keep it localilized patch wise. It'd even let us delete our TreeMap change.
> It just needs `.sorted((a,b) -> a.split(" ")[1].compareTo(b.split(" ")[1])`

Assuming you want that `.sorted((a,b) -> a.split(" ")[1].compareTo(b.split(" ")[1])`
in the same place as you mentioned [here](https://github.com/MinecraftForge/MinecraftForge/issues/6783#issuecomment-647029915), it doesn't seem to like it.
![patch](https://user-images.githubusercontent.com/2760645/85312703-d1fa2f00-b47c-11ea-941d-64e3cd038635.png)

hanetzer commented:
pup helped me sort out the sorting you suggested and it works on both my (linux)
and their (windows) machines with the same output.

hanetzer committed:
MarkDimensionForDeletionTest: update to proper mapping

Signed-off-by: Marty E. Plummer <hanetzer@startmail.com>

hanetzer committed:
DirectoryCache: sort while writing

Allows us to remove the change from HashMap to TreeMap.

Signed-off-by: Marty E. Plummer <hanetzer@startmail.com>

hanetzer commented:
> Your test means nothing. It wouldnt ever expose anything while run on a the same os over and over.

So are you wanting that removed, or what? The reason I put it in is it, without the sorted
snippet inserted, produces different output on windows and linux, so a change which
causes the output to not be normalized the same could be noticed in theory.
